exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/common/HashTableTemplate.java

```diff
@@ -624,6 +639,27 @@ private void retryAfterOOM(boolean batchAdded) throws RetryAfterSpillException {
     throw new RetryAfterSpillException();
   }
 
+  /**
+   *   Return the Hash Value for the row in the Build incoming batch at index:
+   *   (Only apply to Hash Aggregate there's no "Build" side -- only one batch - this one)
+   *
+   * @param incomingRowIdx
+   * @return
+   * @throws SchemaChangeException
+   */
+  @Override
+  public int getBuildHashCodeTreatDataTypes(int incomingRowIdx) throws SchemaChangeException {
```

Comment:
A benefit of this work can be to extract the Hash-Value calculation from the HashTable class(es). Conceptually the two are independent -- the hash-value is a parameter given to the hash table API.